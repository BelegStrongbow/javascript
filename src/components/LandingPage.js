import PropTypes from "prop-types";
import React from "react";
import styled from "styled-components";
import NewTabMessage from "../components/NewTabMessage";

const PageContainer = styled.div`
	margin-top: 10vh;
	text-align: center;

	img {
		width: 540px;
		max-width: 100%;
		height: auto;

		display: block;
		margin: 0 auto;

		@media screen and ( max-width: 400px ) {
			max-width: 250px;
		}
		@media screen and ( min-width: 400px ) {
			max-width: 384px;
		}
	}

	a {
		font-size: 1.5em;
	}
`;

/**
 * Creates the Page  component
 *
 * @param {Object} props The props to use
 *
 * @returns {ReactElement} The rendered component.
 */
export default function LandingPage( props ) {
	return (
		<PageContainer>
			{ props.paragraphs.map( function( paragraph ) {
				return <p key={ paragraph.props.id }>{ paragraph }</p>;
			} ) }
			<p>
				<a href={ props.url } target="_blank" rel="noopener noreferrer">
					{ props.urlText }
					<NewTabMessage />
				</a>
			</p>
			<img src={ props.imageSource } alt="" />
		</PageContainer>
	);
}

LandingPage.propTypes = {
	paragraphs: PropTypes.arrayOf( PropTypes.element ),
	url: PropTypes.string,
	urlText: PropTypes.string,
	imageSource: PropTypes.string,
};
LandingPage.defaultProps = {
	paragraphs: [],
	url: "",
	urlText: "",
	imageSource: "",
};
