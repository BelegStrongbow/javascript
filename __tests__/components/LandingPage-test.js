import React from 'react';
import LandingPage from '../../src/components/LandingPage';
import constructionImage from "../../src/images/construction.svg";
import { createComponentWithIntl } from "../../utils";
import { FormattedMessage } from "react-intl";


test('the landing page component matches the snapshot', () => {
	const component = createComponentWithIntl(
		<LandingPage url="http://yoast.com"
					 urlText="yoast.com"
					 imageSource={ constructionImage }
					 paragraphs={ [ <FormattedMessage id="landing-page.test" defaultMessage="I am a landing page" /> ] }
		/>
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
