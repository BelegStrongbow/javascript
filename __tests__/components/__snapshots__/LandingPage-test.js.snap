// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the landing page component matches the snapshot 1`] = `
.c3 {
  -webkit-transform: translateY(1em);
  -ms-transform: translateY(1em);
  transform: translateY(1em);
}

.c2 {
  display: inline-block;
  height: 48px;
  padding: 12px 16px;
  background-color: #64a60a;
  color: #fff;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2);
  border-radius: 4px;
  vertical-align: top;
  font: 400 14px/24px "Open Sans",sans-serif;
  text-transform: uppercase;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  text-shadow: 0px 0px 2px #000;
  border-bottom: 2px solid rgba(0,0,0,0.2);
  -webkit-transition: background-color 150ms ease-out;
  transition: background-color 150ms ease-out;
}

.c2:hover,
.c2:focus {
  color: #fff;
}

.c2:hover,
.c2:focus {
  box-shadow: 0 4px 10px 0 rgba(0,0,0,0.2),inset 0 0 0 100px rgba( 0,0,0,0.1 );
}

.c2:active {
  -webkit-transform: translateY( 1px );
  -ms-transform: translateY( 1px );
  transform: translateY( 1px );
  box-shadow: none;
}

.c1 {
  min-width: 152px;
}

.c0 {
  margin-top: 10vh;
  text-align: center;
}

.c0 img {
  width: 540px;
  max-width: 100%;
  height: auto;
  display: block;
  margin: 0 auto;
}

@media screen and ( max-width:400px ) {
  .c0 img {
    max-width: 250px;
  }
}

@media screen and ( min-width:400px ) {
  .c0 img {
    max-width: 384px;
  }
}

<div
  className="c0"
>
  <p>
    <span>
      I am a landing page
    </span>
  </p>
  <p>
    <a
      className="c1 c2"
      href="http://yoast.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      yoast.com
      <span
        className="screen-reader-text c3"
      >
        <span>
          (Opens in a new browser tab)
        </span>
      </span>
    </a>
  </p>
  <img
    alt=""
    src="construction.svg"
  />
</div>
`;
